packages/cli/src/commands/build.ts

```diff
@@ -35,10 +35,27 @@ export default class Build extends Command {
 
     // Get the output location from the project package.json main field
     const pjson = JSON.parse(readFileSync(path.join(directory, 'package.json')).toString());
-    const outputPath = path.resolve(directory, pjson.main || 'dist/index.js');
+
+    let buildEntries;
+    const defaultEntry = path.join(directory, 'src/index.ts');
+
+    if (pjson.exports) {
+      buildEntries = pjson.exports;
+      const name = path.basename(defaultEntry, '.js');
+      buildEntries[name] = defaultEntry;
+    } else {
+      buildEntries = defaultEntry;
+    }
+
+    for (const i in buildEntries) {
+      if (typeof buildEntries[i] !== 'string') {
+        this.warn(`ignoring nested field ${i} from build`);
+        delete buildEntries[i];
+      }
```

Comment:
Webpack can also support `string[] | EntryDescription` as values. I need to add these.